kv/kvserver: TestMergeQueue failed

kv/kvserver.TestMergeQueue [failed](https://mesolite.cluster.engflow.com/invocations/default/b663b449-1d95-4866-bb40-611cb328721d?testReportRun=3&testReportShard=23&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on release-26.1.0-rc @ [3b61854ae28ed972d6f26a2b72f24bf8fb513e42](https://github.com/cockroachdb/cockroach/commits/3b61854ae28ed972d6f26a2b72f24bf8fb513e42):

Fatal error:

```
panic: runtime error: invalid memory address or nil pointer dereference
	(1) attached stack trace
	  -- stack trace:
	  | runtime.gopanic
	  | 	GOROOT/src/runtime/panic.go:783
	  | github.com/cockroachdb/cockroach/pkg/util/ctxgroup.Group.Wait
	  | 	pkg/util/ctxgroup/ctxgroup.go:151
	  | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.muxRangeFeed
	  | 	pkg/kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:94
	  | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).RangeFeed
	  | 	pkg/kv/kvclient/kvcoord/dist_sender_rangefeed.go:207
	  | github.com/cockroachdb/cockroach/pkg/kv/kvclient/rangefeed.(*dbAdapter).RangeFeed
	  | 	pkg/kv/kvclient/rangefeed/db_adapter.go:82
	  | github.com/cockroachdb/cockroach/pkg/kv/kvclient/rangefeed.(*RangeFeed).run.func1
	  | 	pkg/kv/kvclient/rangefeed/rangefeed.go:370
	  | [...repeated from below...]
	Wraps: (2) attached stack trace
	  -- stack trace:
	  | runtime.gopanic
	  | 	GOROOT/src/runtime/panic.go:783
	  | runtime.panicmem
	  | 	GOROOT/src/runtime/panic.go:262
	  | runtime.sigpanic
	  | 	GOROOT/src/runtime/signal_unix.go:925
	  | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*activeMuxRangeFeed).start
	  | 	pkg/kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:285
	  | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*rangefeedMuxer).restartActiveRangeFeed
	  | 	pkg/kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:587
	  | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*rangefeedMuxer).receiveEventsFromNode.func1
	  | 	pkg/kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:514
	  | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*rangefeedMuxer).receiveEventsFromNode.Group.GoCtx.func2
	  | 	pkg/util/ctxgroup/ctxgroup.go:200
	  | golang.org/x/sync/errgroup.(*Group).Go.func1
	  | 	external/org_golang_x_sync/errgroup/errgroup.go:93
	  | runtime.goexit
	  | 	src/runtime/asm_amd64.s:1693
	Wraps: (3) runtime error: invalid memory address or nil pointer dereference
	Error types: (1) *withstack.withStack (2) *withstack.withStack (3) runtime.errorString

```
Stack: 

```
goroutine 38715 [running]:
github.com/cockroachdb/cockroach/pkg/util/ctxgroup.Group.Wait({0xc004441640?, {0xddeade0?, 0xc0021094a0?}, 0xc0096e2ca0?})
	pkg/util/ctxgroup/ctxgroup.go:151 +0x134
github.com/cockroachdb/cockroach/pkg/util/ctxgroup.GoAndWait({0xddeae18, 0xc0093fc5d0}, {0xc00b721b80, 0x2, 0xb74b580?})
	pkg/util/ctxgroup/ctxgroup.go:224 +0x1b9
github.com/cockroachdb/cockroach/pkg/kv/kvclient/rangefeed.(*RangeFeed).run(0xc008386a80, {0xddeae18, 0xc0093fc5d0}, {0xde53dc0, 0xc00d431e00}, 0x0)
	pkg/kv/kvclient/rangefeed/rangefeed.go:396 +0x12ee
github.com/cockroachdb/cockroach/pkg/kv/kvclient/rangefeed.(*RangeFeed).start.func1({0xddeade0, 0xc00d431e50})
	pkg/kv/kvclient/rangefeed/rangefeed.go:256 +0x43f
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func1({0xddeade0, 0xc00d431e50})
	pkg/util/stop/stopper.go:440 +0x112
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx in goroutine 35921
	pkg/util/stop/stopper.go:438 +0x247
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestMergeQueue
    test_log_scope.go:171: test logs captured to: outputs.zip/logTestMergeQueue3579780438
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:668: DRPC is enabled (via override by TestingGlobalDRPCOption)
=== RUN   TestMergeQueue/sticky-bit-expiration
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=3</code>
 - <code>shard=23</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv-triage
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMergeQueue.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-59574

--- a/pkg/server/server.go
+++ b/pkg/server/server.go
@@ -930,7 +930,6 @@ func NewServer(cfg Config, stopper *stop.Stopper) (serverctl.ServerStartupInterf
 			}
 			storeLoadMsg := mmaintegration.MakeStoreLoadMsg(storeDesc, origTimestampNanos)
 			mmaAlloc.SetStore(state.StoreAttrAndLocFromDesc(storeDesc))
-			ctx = logtags.AddTag(ctx, "s", storeDesc.StoreID)
 			mmaAlloc.ProcessStoreLoadMsg(ctx, &storeLoadMsg)
 		},
 	)

